src/software/ai/hl/stp/play/kickoff_enemy_play_test.cpp

```diff
@@ -28,18 +31,62 @@ TEST_F(KickoffEnemyPlayTest, test_kickoff_enemy_play)
         [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
-            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(9.5))
+            auto robotsShadowingEnemy = [](std::shared_ptr<World> world_ptr) -> bool {
+                // TODO: Fix bug with robot three not shadowing the enemy kicker in
+                // kickoff_enemy_play
+                // https://github.com/UBC-Thunderbots/Software/issues/1945
+
+                Rectangle robotOneShadowingRect(Point(0, 2.2), Point(-0.4, 1.8));
+                Rectangle robotFiveShadowingRect(Point(0, -2.2), Point(-0.4, -1.8));
+
+                Point robotOnePos =
+                    world_ptr->friendlyTeam().getRobotById(1).value().position();
+                Point robotFivePos =
+                    world_ptr->friendlyTeam().getRobotById(5).value().position();
+
+                return contains(robotOneShadowingRect, robotOnePos) &&
+                       contains(robotFiveShadowingRect, robotFivePos);
+            };
+
+            auto robotsDefendingPosts = [](std::shared_ptr<World> world_ptr) -> bool {
+                // Positions taken from kickoff_enemy_play
+                Point robotTwoExpectedPos =
+                    Point(world_ptr->field().friendlyGoalpostPos().x() +
+                              world_ptr->field().defenseAreaXLength() +
+                              2 * ROBOT_MAX_RADIUS_METERS,
+                          world_ptr->field().defenseAreaYLength() / 2.0);
+                Point robotFourExpectedPos =
+                    Point(world_ptr->field().friendlyGoalpostNeg().x() +
+                              world_ptr->field().defenseAreaXLength() +
+                              2 * ROBOT_MAX_RADIUS_METERS,
+                          -world_ptr->field().defenseAreaYLength() / 2.0);
+
+                double tolerance = 0.05;
+
+                Circle robotTwoCircle(robotTwoExpectedPos, tolerance);
+                Circle robotFourCircle(robotFourExpectedPos, tolerance);
+
+                Point robotTwoPos =
+                    world_ptr->friendlyTeam().getRobotById(2).value().position();
+                Point robotFourPos =
+                    world_ptr->friendlyTeam().getRobotById(4).value().position();
+
+                return contains(robotTwoCircle, robotTwoPos) &&
+                       contains(robotFourCircle, robotFourPos);
+            };
+
+            while (!robotsDefendingPosts(world_ptr) || !robotsShadowingEnemy(world_ptr))
             {
                 yield();
             }
         }};
 
-    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+    std::vector<ValidationFunction> non_terminating_validation_functions = {
+        [](std::shared_ptr<World> world_ptr, ValidationCoroutine::push_type& yield) {
+            robotsInFriendlyHalf(world_ptr, yield);
+            robotsNotInCenterCircle(world_ptr, yield);
```

Comment:
```suggestion
           robotsInFriendlyHalf, robotsNotInCenterCircle
```
should these be run in parallel?

Reply:
I'm not sure if it would make any noticeable difference. I'm also not sure how I could have two functions running in parallel.

Reply:
seems like Akhil agrees, so bumping this. The suggested change should make it go in parallel

Reply:
Updated!